lib/yard/parser/c/c_parser.rb

```diff
@@ -38,6 +38,7 @@ def parse_toplevel
             when '#'; consume_directive
             when '/'; consume_comment
             when /\s/; consume_whitespace
+            when '}'; advance # Skip possible C++ namespace closing brackets.
```

Comment:
So this one, I'm not sure if you'll like this much. I can try to refactor it if you prefer. Though any suggestions would be appreciated. The issue here was that when there are namespaces my fix was to just ignore it. But then `consume_toplevel_statement` would later run into stray `}` (closing namespaces) that would throw it off.

So this line is currently very generic in advancing over any `}` in the top level. I could try to keep track of the depth of the C++ namespaces and only advance over `}` a matching number of times.

It all depend on how low impact we want to keep this. I ran this fix over our code base with over 1300 methods and 98 classes - no documentation objects where lost.